File: inspect_minna_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

const pdfPath = 'TỪ VỰNG 50 BÀI MINNA NO NIHONGO (2).pdf';

let dataBuffer = fs.readFileSync(pdfPath);

pdf(dataBuffer).then(function(data) {
    console.log("Total Pages:", data.numpages);
    console.log("Info:", data.info);
    console.log("Text length:", data.text.length);
    console.log("\n--- FIRST 2000 CHARACTERS ---");
    console.log(data.text.slice(0, 2000));
    
    // Write sample text to file for detailed inspection
    fs.writeFileSync('minna_extracted_sample.txt', data.text.slice(0, 10000), 'utf8');
    console.log("\nSaved first 10,000 characters to minna_extracted_sample.txt");
}).catch(err => {
    console.error("PDF parse error:", err);
});

File: inspect_minna_pdf.py
import pypdf
import sys

pdf_path = "TỪ VỰNG 50 BÀI MINNA NO NIHONGO (2).pdf"

try:
    reader = pypdf.PdfReader(pdf_path)
    print(f"Total pages: {len(reader.pages)}")
    text_sample = ""
    for i in range(min(5, len(reader.pages))):
        page_text = reader.pages[i].extract_text()
        print(f"--- PAGE {i+1} ---")
        print(page_text[:500])
except Exception as e:
    print(f"pypdf error: {e}")
    # try pdfplumber if available
    try:
        import pdfplumber
        with pdfplumber.open(pdf_path) as pdf:
            print(f"pdfplumber total pages: {len(pdf.pages)}")
            for i in range(min(3, len(pdf.pages))):
                print(f"--- PLUMBER PAGE {i+1} ---")
                print(pdf.pages[i].extract_text()[:500])
    except Exception as e2:
        print(f"pdfplumber error: {e2}")
